toucan_connectors/mongo/mongo_connector.py

```diff
@@ -24,23 +24,35 @@
 MAX_COUNTED_ROWS = 1000001
 
 
-def _is_match_empty(match_: dict) -> bool:
-    def is_empty(elem: Any):
-        # Not matching None here, because we don't want to consider {'$match': {'col': None}} to be
-        # empty
-        if elem == {}:
-            return True
-        if isinstance(elem, dict):
-            return _is_match_empty(elem)
-        return False
+def _is_empty_match_column(elem: Any):
+    # Not matching None here, because we don't want to consider {'$match': {'col': None}} to be
+    # empty
+    if elem == {}:
```

Comment:
```suggestion
    if not bool(elem):
```

```bash
In [1]: not bool({})
Out[1]: True

In [2]: not bool([])
Out[2]: True
```
WDYT ?
```

Reply:
We can't really do that since: 
```
>>> not bool(None)
True
```

But we can do 
```suggestion
    if elem == {} or elem == []:
```

Reply:
Fair enough 👌🏽

Reply:
Finalement, je pense qu'il ne faut pas rajouter le check sur le `[]` parce qu'on ne veut pas virer les steps où on a par exemple: `{'some_column': {'$in': []}`. ça va donner un résultat qui est faux.